### PR TITLE
feat(world): deployment salt by msg.sender

### PR DESCRIPTION
**Pull request by yonadaaa**
Fixes [#1850](https://github.com/latticexyz/mud/issues/1850)

### CODE REVIEW

File: packages/world/src/WorldFactory.sol [hunk 1]
```diff
@@ -17,8 +17,8 @@ contract WorldFactory is IWorldFactory {
   /// @notice Address of the core module to be set in the World instances.
   IModule public immutable coreModule;
 
-  /// @notice Counter to keep track of the number of World instances deployed.
-  uint256 public worldCount;
+  /// @notice Counters to keep track of the number of World instances deployed per address.
+  mapping(address => uint256) public accountCount;
```
Comment (holic) [inline packages/world/src/WorldFactory.sol:21]: we can name mapping params now, which helps to self-document
```suggestion
  mapping(address creator => uint256 worldCount) public accountCount;
```
not blocking, but I kinda feel like `worldCounts` might be clearer than `accountCount`?

File: .changeset/selfish-pears-marry.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@latticexyz/world": minor
+---
+
+Added `msg.sender` to the `WorldFactory` deployment salt, so the World address is derived from the creator of the world.
```
Comment (holic) [inline .changeset/selfish-pears-marry.md:5]:
```suggestion
`WorldFactory` now derives a salt based on number of worlds deployed by `msg.sender`, which should help with deterministic world deployments across chains.
```

**Review by holic — DISMISSED** [review]

**Comment by holic** [comment]
do we need to regenerate test data after this change?

File: packages/world/src/WorldFactory.sol (hunk 1 above)
Comment (yonadaaa) [reply packages/world/src/WorldFactory.sol:21]: I agree, then it's like "world counts per account"

**Comment by yonadaaa** [comment]
> do we need to regenerate test data after this change?

It might not be strictly necessary (if nothing relies on addresses) but I'll do it anyway

**Review by holic — APPROVED** [review]

**Pull request merged**